fvm/evm/offchain/blocks/blocks.go

```diff
@@ -0,0 +1,122 @@
+package blocks
+
+import (
+	"fmt"
+
+	gethCommon "github.com/onflow/go-ethereum/common"
+
+	"github.com/onflow/flow-go/fvm/evm/handler"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const BlockStoreLatestBlockMetaKey = "LatestBlockMeta"
+
+// Blocks facilitates access to the recent block hash values
+// and also the latest executed block meta data
+type Blocks struct {
+	chainID     flow.ChainID
+	storage     types.BackendStorage
+	rootAddress flow.Address
+	bhl         *handler.BlockHashList
+}
+
+// NewBlocks constructs a new blocks type
+func NewBlocks(
+	chainID flow.ChainID,
+	rootAddress flow.Address,
+	storage types.BackendStorage,
+) (*Blocks, error) {
+	var err error
+	blocks := &Blocks{
+		chainID:     chainID,
+		storage:     storage,
+		rootAddress: rootAddress,
+	}
+	blocks.bhl, err = handler.NewBlockHashList(
+		storage,
+		rootAddress,
+		handler.BlockHashListCapacity,
+	)
+	if err != nil {
+		return nil, err
+	}
+	// if empty insert genesis block hash
+	if blocks.bhl.IsEmpty() {
+		genesis := types.GenesisBlock(chainID)
+		err = blocks.PushBlockMeta(
+			NewMeta(
```

Comment:
Would it make sense to also include the `ParentHash`? I don't have a specific use-case in mind, other than providing a way to verify that the block hash list has no gaps.

Reply:
We can fetch the `parentHash` from the hashlist if needed